tests/jobs/nosim/test_nosim.py Fail after flipping config-parser flag

tests/jobs/nosim/test_nosim.py fail after flipping flag running the new config-parser. Problem seems to be related to `INSERT_NOSIM`

```
   @pytest.mark.skipif(
        sys.platform == "darwin", reason="Skip test for Mac OS - invalid use of sed"
    )
    # The GNU extension for the sed command on osx behaves slightly different.
    # The result when running the forward model job is as follow:
    # cat insert_nosim.stderr.1
    # sed: 1: "TEST.DATA": invalid command code T
    @pytest.mark.parametrize(
        "nosim_command,data_input,data_expected",
        [
            (
                "INSERT_NOSIM",
                "RUNSPEC\n",
                "RUNSPEC\nNOSIM\n",
            ),
            (
                "INSERT_NOSIM",
                "RUNSPEC -- some comment about RUNSPEC and whatnot\n-- and more RUNSPEC\n",
                "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
            ),
            (
                "INSERT_NOSIM",
                "RUNSPEC some comment about RUNSPEC and whatnot\n-- and more RUNSPEC\n",
                "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
            ),
            (
                "REMOVE_NOSIM",
                "RUNSPEC\nNOSIM\n",
                "RUNSPEC\n",
            ),
            (
                "REMOVE_NOSIM",
                "RUNSPEC\nNOSIM\n-- and more RUNSPEC\n",
                "RUNSPEC\n-- and more RUNSPEC\n",
            ),
            (
                "REMOVE_NOSIM",
                (
                    "RUNSPEC for some reason valid -- some comment RUNSPEC\n"
                    "-- more\n"
                    "NOSIM   more valid comment because it is after 8 chars"
                    " -- and now a valid comment\n"
                    "-- and a comment about NOSIM as well, not to be touched\n"
                ),
                (
                    "RUNSPEC for some reason valid -- some comment RUNSPEC\n"
                    "-- more\n-- and a comment about NOSIM as well, not to be touched\n"
                ),
            ),
        ],
    )
    @pytest.mark.usefixtures("setup_tmpdir")
    def test_nosim(nosim_command, data_input, data_expected):
        shutil.copy(os.path.join(os.path.dirname(__file__), "data", "nosim.ert"), ".")
    
        with open("nosim.ert", "a", encoding="utf-8") as fh:
            fh.writelines([f"FORWARD_MODEL {nosim_command}"])
    
        with open("TEST.DATA", "w", encoding="utf-8") as fh:
            fh.write(data_input)
    
        with ErtPluginContext(
            plugins=[semeio.hook_implementations.jobs, ert.shared.hook_implementations]
        ):
            subprocess.check_call(
                ["ert", "test_run", "nosim.ert", "--verbose"],
            )
        with open("nosim/real_0/iter_0/TEST.DATA", encoding="utf-8") as fh:
>           assert fh.read() == data_expected
E           AssertionError: assert 'RUNSPEC\n' == 'RUNSPEC\nNOSIM\n'
E               RUNSPEC
E             - NOSIM
```

--- a/src/ert/parsing/file_context_token.py
+++ b/src/ert/parsing/file_context_token.py
@@ -34,6 +34,17 @@ class FileContextToken(Token):
     def __str__(self) -> str:
         return self.value
 
+    def __eq__(self, other) -> bool:
+        if isinstance(other, str):
+            return self.value == other
+        if isinstance(other, Token):
+            return self.value == other.value
+        else:
+            return False
+
+    def __hash__(self):
+        return hash(self.value)
+
     @classmethod
     def join_tokens(
         cls, tokens: List["FileContextToken"], separator: str = " "

--- a/tests/test_config_parsing/test_forward_model.py
+++ b/tests/test_config_parsing/test_forward_model.py
@@ -410,3 +410,29 @@ def test_that_spaces_in_forward_model_args_are_dropped():
     assert len(ert_config.forward_model_list) == 1
     job = ert_config.forward_model_list[0]
     assert job.private_args.get("<VERSION>") == "smersion"
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_forward_model_with_different_token_kinds_are_added():
+    """
+    This is a regression tests for a problem where the parser had different
+    token kinds which ended up in separate keys in the input dictionary, and were
+    therefore not added
+    """
+    test_config_file_name = "test.ert"
+    Path("job").write_text("EXECUTABLE echo\n", encoding="utf-8")
+    test_config_contents = dedent(
+        """
+        NUM_REALIZATIONS 1
+        INSTALL_JOB job job
+        FORWARD_MODEL job
+        FORWARD_MODEL job(<MESSAGE>=HELLO)
+        """
+    )
+    with open(test_config_file_name, "w", encoding="utf-8") as fh:
+        fh.write(test_config_contents)
+
+    assert [
+        (j.name, len(j.private_args))
+        for j in ErtConfig.from_file(test_config_file_name).forward_model_list
+    ] == [("job", 0), ("job", 1)]

